test/units/cli/test_galaxy.py

```diff
@@ -444,3 +460,222 @@ def test_template_ignore_similar_folder(self):
+    assert metadata['readme'] == 'README.md'
+    assert metadata['version'] == '1.0.0'
+    assert metadata['description'] == 'your description'
+    assert metadata['license'] == 'license (GPL-2.0-or-later, MIT, etc)'
+    assert metadata['tags'] == []
+    assert metadata['dependencies'] == {}
+    assert metadata['documentation'] == 'http://docs.example.com'
+    assert metadata['repository'] == 'http://example.com/repository'
+    assert metadata['homepage'] == 'http://example.com'
+    assert metadata['issues'] == 'http://example.com/issue/tracker'
+    assert len(metadata) == 13
+
+    for d in ['docs', 'plugins', 'roles']:
+        assert os.path.isdir(os.path.join(collection_skeleton, d)), \
+            "Expected collection subdirectory {0} doesn't exist".format(d)
+
+
+@pytest.mark.parametrize('collection_skeleton', [
+    ('ansible_test.delete_me_skeleton', os.path.join(os.path.split(__file__)[0], 'test_data', 'collection_skeleton')),
+], indirect=True)
+def test_collection_skeleton(collection_skeleton):
+    meta_path = os.path.join(collection_skeleton, 'galaxy.yml')
+
+    with open(meta_path, 'r') as galaxy_meta:
+        metadata = yaml.safe_load(galaxy_meta)
+
+    assert metadata['namespace'] == 'ansible_test'
+    assert metadata['name'] == 'delete_me_skeleton'
+    assert metadata['authors'] == ['Ansible Cow <acow@bovineuniversity.edu>', 'Tu Cow <tucow@bovineuniversity.edu>']
+    assert metadata['version'] == '0.1.0'
+    assert metadata['readme'] == 'README.md'
+    assert len(metadata) == 5
+
+    assert os.path.exists(os.path.join(collection_skeleton, 'README.md'))
+
+    # Test empty directories exist and are empty
+    for empty_dir in ['plugins/action', 'plugins/filter', 'plugins/inventory', 'plugins/lookup',
+                      'plugins/module_utils', 'plugins/modules']:
+
+        assert os.listdir(os.path.join(collection_skeleton, empty_dir)) == []
+
+    # Test files that don't end with .j2 were not templated
+    doc_file = os.path.join(collection_skeleton, 'docs', 'My Collection.md')
+    with open(doc_file, 'r') as f:
+        doc_contents = f.read()
+    assert doc_contents.strip() == 'Welcome to my test collection doc for {{ namespace }}.'
+
+    # Test files that end with .j2 but are in the templates directory were not templated
+    for template_dir in ['playbooks/templates', 'playbooks/templates/subfolder',
+                         'roles/common/templates', 'roles/common/templates/subfolder']:
+        test_conf_j2 = os.path.join(collection_skeleton, template_dir, 'test.conf.j2')
+        assert os.path.exists(test_conf_j2)
+
+        with open(test_conf_j2, 'r') as f:
+            contents = f.read()
+        expected_contents = '[defaults]\ntest_key = {{ test_variable }}'
+
+        assert expected_contents == contents.strip()
+
+
+@pytest.fixture()
+def collection_build(collection_skeleton, tmp_path_factory):
+    output_dir = str(tmp_path_factory.mktemp('test-ÅÑŚÌβŁÈ Output'))
```

Comment:
Why did you choose to use strings instead of a `Path` instance?
The best practice is to use the object representation and only turn it to string in places where you pass it into functions which don't support the fspath protocol. Like stringify it in that `galaxy_args` list but `yield` the original `Path` object which then later could be used nicely down the control flow.

Reply:
Mainly because I didn't want to deal with stringifying it when it was needed. Honestly I just prefer it this way and because it's a test I don't really want to deal with the other way. Call it lazy but I prefer just dealing with a string than the object.